Use mksquashfs instead of downloading during the build

Mailaender opened the pull request:
~~Required for https://github.com/AppImage/appimaged/issues/40.~~ Actually I am wrong. That silently moved to https://github.com/AppImage/appimaged. This change allows the AppImageKit to compile on @openSUSE build servers which disallow remote code downloading during the build for security reasons and the dependency should be unbundled anyway.

TheAssassin commented on cmake/dependencies.cmake:
```diff
@@ -30,7 +30,9 @@ set(CFLAGS ${DEPENDENCIES_CFLAGS})
 
+if(NOT USE_SYSTEM_MKSQUASHFS)
 # TODO: allow using system wide mksquashfs
 set(mksquashfs_cflags "-DXZ_SUPPORT ${CFLAGS}")
```
Indentation needs to be adjusted within the first `if` branch.

Mailaender commented:
> I don't see the relation to that appimaged issue. Please explain your intent more detailedly.

It seems that this is a bundled dependency of appimaged. I am a bit confused as well.

TheAssassin commented:
AppImageKit is used to build appimaged AppImages, but it's not a runtime dependency of any kind.

probonopd commented:
@Mailaender I can only speculate as to the motivation for this PR, but if I understand it correctly, it is because OBS does not allow to access the internet (and download additional sources) at build time. Is this the rationale for this PR?

Mailaender commented:
Yes.

Mailaender commented:
Updated with more changes from https://build.opensuse.org/package/view_file/home:munix9:unstable/AppImageKit/fix-disable-mksquashfs-build.patch and changed the indentions.

Mailaender commented:
Actually had to remove that again and it can be solved by emptying `AUXILIARY_FILES_DESTINATION` I suppose.

Mailaender commented:
Updated again to fix the installation.

probonopd commented:
Something is going wrong:

```
-- Set runtime path of "install_prefix//usr/lib64/libappimage.so.0.1.7" to ""

-- Installing: install_prefix/usr/include/appimage/appimage.h

CMake Error at src/cmake_install.cmake:41 (file):

  file INSTALL cannot find "/AppImageKit/build/mksquashfs-prefix/mksquashfs

  /AppImageKit/build/src/runtime".

Call Stack (most recent call first):

  cmake_install.cmake:43 (include)
```

Do you have a suspicion what is going wrong? Unfortunately I have no clue about CMake, but maybe @TheAssassin has an idea.

Mailaender commented:
Fixed it. Sorry.

TheAssassin commented on cmake/dependencies.cmake:
```diff
@@ -30,49 +30,55 @@ set(CFLAGS ${DEPENDENCIES_CFLAGS})
 
+set(USE_SYSTEM_MKSQUASHFS OFF CACHE BOOL "Use system mksquashfs instead of downloading and building our own")
+
+if(NOT USE_SYSTEM_MKSQUASHFS)
```
I'd like to see a warning that the system `mksquashfs` might not work as intended. Only the most recent version *might* work.

Mailaender replied:
Added a warning.

probonopd commented:
Since @TheAssassin and Travis CI both seem to like it, so do I :+1:
